app/src/util/pdf_utils.py

```diff
@@ -0,0 +1,60 @@
+from dataclasses import dataclass
+from typing import Any
+
+from pdfminer.pdfdocument import PDFDocument, PDFNoOutlines
+from pdfminer.pdfpage import PDFPage
+from pdfminer.pdftypes import PDFObjRef, resolve1
+from pdfminer.psparser import PSLiteral
+
+
+@dataclass
+class Heading:
+    title: str
+    # Heading level starting from 1
+    level: int
+    # Page number where the heading first appears
+    pageno: int | None
```

Comment:
Alternatively, since we don't yet have a use case for document_page_number, I can also revise my PR to remove the Page class and instead have just an int -- I slightly prefer this option, but LMK either way and I'll update my PR accordingly before merging!

Reply:
Sounds good for both comments. We start with `int` and create a `Page` class when needed.

Reply:
Will make this change in mine now!